app/controllers/finance_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class FinanceController < ApplicationController
+  before_action :find_finance, only: %i[show edit update destroy]
+  before_action :require_same_user, only: %i[edit update destroy info]
+
+  def new
+    @finance = Finance.new
+  end
+
+  def edit; end
+
+  def update
+    if @finance.update(finance_params)
+      PersonsFinance.where(finance_id: @finance.id).destroy_all
+      params[:finance][:id].each do |person_id|
```

Comment:
please use your params with `params.require(:finance).permit(:id)
u can create new method for this cuz it can be reused in oher places